src/rand_impls.rs

```diff
@@ -176,6 +207,94 @@ impl Rand for bool {
     }
 }
 
+impl Rand for Ordering {
+    #[inline]
+    fn rand<R: Rng>(rng: &mut R) -> Ordering {
+        match rng.gen_range(0, 3) {
+            0 => Ordering::Less,
+            1 => Ordering::Equal,
+            2 => Ordering::Greater,
+            _ => unreachable!(),
+        }
+    }
+}
+
+#[cfg(feature = "duration")]
+impl Rand for Duration {
+    #[inline]
+    fn rand<R: Rng>(rng: &mut R) -> Duration {
+        Duration::new(rng.gen(), rng.gen_range(0, 1_000_000_000))
+    }
+}
+
+impl Rand for RangeFull {
+    #[inline]
+    fn rand<R: Rng>(rng: &mut R) -> RangeFull {
+        ..
+    }
+}
+
+impl<T: Rand + PartialOrd> Rand for Range<T> {
+    #[inline]
+    fn rand<R: Rng>(rng: &mut R) -> Range<T> {
+        Range { start: rng.gen(), end: rng.gen() }
+    }
+}
+
+impl<T: Rand> Rand for RangeFrom<T> {
+    #[inline]
+    fn rand<R: Rng>(rng: &mut R) -> RangeFrom<T> {
+        RangeFrom { start: rng.gen() }
+    }
+}
+
+impl<T: Rand> Rand for RangeTo<T> {
+    #[inline]
+    fn rand<R: Rng>(rng: &mut R) -> RangeTo<T> {
+        RangeTo { end: rng.gen() }
+    }
+}
+
+#[cfg(feature = "bound")]
+impl<T: Rand> Rand for Bound<T> {
+    #[inline]
+    fn rand<R: Rng>(rng: &mut R) -> Bound<T> {
+        match rng.gen_range(0, 3) {
+            0 => Bound::Included(rng.gen()),
+            1 => Bound::Excluded(rng.gen()),
+            2 => Bound::Unbounded,
+            _ => unreachable!(),
+        }
+    }
+}
+
+impl Rand for Ipv4Addr {
+    #[inline]
+    fn rand<R: Rng>(rng: &mut R) -> Ipv4Addr {
```

Comment:
`rng.gen::<i8>()` would prevent overflow

Reply:
Should this exclude certain special addresses perhaps, e.g. those [ending in 0 or 255](https://en.wikipedia.org/wiki/IPv4#Addresses_ending_in_0_or_255)?

May be easiest and fastest to implement via looping on rejection.